### PR TITLE
Split Sdk.targets and update Feed Task project to use `dotnet pack`

### PR DESCRIPTION
**Pull request by alexperovich**
This change splits the "BuildTask" part out of Sdk.targets and updates the Feed Task package to use it.

### CODE REVIEW

File: src/Microsoft.DotNet.Build.Tasks.Feed/Microsoft.DotNet.Build.Tasks.Feed.csproj [hunk 1]
```diff
@@ -2,67 +2,23 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>netstandard1.5;$(NetFxTfm)</TargetFrameworks>
+    <TargetFrameworks>netstandard2.0;$(NetFxTfm)</TargetFrameworks>
```
Comment (JohnTortugo) [inline src/Microsoft.DotNet.Build.Tasks.Feed/Microsoft.DotNet.Build.Tasks.Feed.csproj]: I'm wondering if there is any issue in upgrading this version?

File: src/Microsoft.DotNet.Build.Tasks.Feed/Microsoft.DotNet.Build.Tasks.Feed.csproj [hunk 2]
```diff
@@ -2,67 +2,23 @@
...
 
     <Description>This package provides support for publishing assets to a nuget protocol based feed.</Description>
-  </PropertyGroup>
-
-  <PropertyGroup Condition="'$(TargetFramework)' == '$(NetFxTfm)'">
-      <NuGetRuntimeIdentifier>win7-x64</NuGetRuntimeIdentifier>
-      <PackageTargetFallback>$(PackageTargetFallback);net46</PackageTargetFallback>
-  </PropertyGroup>
-
-  <PropertyGroup Condition="'$(TargetFramework)' == 'netstandard1.5'">
```
Comment (JohnTortugo) [inline src/Microsoft.DotNet.Build.Tasks.Feed/Microsoft.DotNet.Build.Tasks.Feed.csproj]: Change this to netstandard2.0 ?

File: src/Microsoft.DotNet.Build.Tasks.Feed/Microsoft.DotNet.Build.Tasks.Feed.csproj (hunk 1 above)
Comment (weshaggard) [reply src/Microsoft.DotNet.Build.Tasks.Feed/Microsoft.DotNet.Build.Tasks.Feed.csproj]: We have to move it to a netstandard20+ TFM with the latest msbuild packages. However with that said I think for most of the newer tasks we have been skipping netstandard and just targeting netcoreapp2.0 and net461.

Comment (alexperovich) [reply src/Microsoft.DotNet.Build.Tasks.Feed/Microsoft.DotNet.Build.Tasks.Feed.csproj]: The Microsoft.Build package versions we are using in Versions.props don't support netstandard1.5.

File: src/Microsoft.DotNet.Build.Tasks.Feed/Microsoft.DotNet.Build.Tasks.Feed.csproj (hunk 2 above)
Comment (alexperovich) [reply src/Microsoft.DotNet.Build.Tasks.Feed/Microsoft.DotNet.Build.Tasks.Feed.csproj]: These properties were unnecessary.

File: eng/Sdk.targets [hunk 3]
```diff
@@ -5,57 +5,7 @@
 
   <PropertyGroup>
     <PackageType>MSBuildSdk</PackageType>
```
Comment (weshaggard) [inline eng/Sdk.targets]: Given this is the only property here now I wonder if we should just have the BuildTasks.targets imported by all our projects and folks that want to be an SDK set this one property.

File: src/Microsoft.DotNet.Build.Tasks.Feed/Microsoft.DotNet.Build.Tasks.Feed.csproj (hunk 2 above)
Comment (JohnTortugo) [reply src/Microsoft.DotNet.Build.Tasks.Feed/Microsoft.DotNet.Build.Tasks.Feed.csproj]: Sorry didn't see it was a removal.

File: eng/Sdk.targets (hunk 3 above)
Comment (alexperovich) [reply eng/Sdk.targets]: That's fair. I will remove it.

**Comment by alexperovich** [comment]
Filed #393 to track deciding on TFMs.

**Pull request merged**